enumerables.rb

```diff
@@ -0,0 +1,152 @@
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    n = 0
+    while n < size
+      yield(self[n])
+      n += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    n = 0
+    x = 0
+    while n < size
+      yield(self[n], x)
+      n += 1
+      x += 1
+
+    end
+    self
+  end
```

Comment:
## Hashes, range and array self
- [ ] Test with hashes and ranges, then compare with original `each_with_index` Ruby method
- [ ] Fix your method to work with the tests. Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)

**Example:**

`test_file.rb`
```ruby
(0..9).my_each_with_index{
# or
Range.new(0, 9)

# HASHES
{ a: 1, b: 2, c: 3, d: 4, e: 5 }
```